src/quicr_client.cpp

```diff
@@ -248,21 +434,30 @@ QuicRClient::handle(messages::MessageBuffer&& msg)
       std::cout << "Got SubscribeResponse: No delegate found for namespace"
                 << response.quicr_namespace.to_hex() << std::endl;
     }
+
   } else if (msg_type == static_cast<uint8_t>(messages::MessageType::Publish)) {
     messages::PublishDatagram datagram;
     msg >> datagram;
-    for (const auto& entry : sub_delegates) {
-      if (entry.first.contains(datagram.header.name)) {
-        sub_delegates[entry.first]->onSubscribedObject(
-          datagram.header.name,
-          0x0,
-          0x0,
-          false,
-          std::move(datagram.media_data));
-      } else {
-        std::cout << "Name:" << datagram.header.name.to_hex()
-                  << ", not in namespace " << entry.first.to_hex() << std::endl;
+
+    if (from_varint(datagram.header.offset_and_fin) == 0x1) {
+      // No-fragment, process as single object
+
+      for (const auto& entry : sub_delegates) {
+        if (entry.first.contains(datagram.header.name)) {
+          sub_delegates[entry.first]->onSubscribedObject(
+            datagram.header.name,
+            0x0,
+            0x0,
+            false,
+            std::move(datagram.media_data));
+        } else {
+          std::cout << "Name:" << datagram.header.name.to_hex()
+                    << ", not in namespace " << entry.first.to_hex()
+                    << std::endl;
```

Comment:
Is this a helpful output? Shouldn't we output only if we never found a valid namespace?

Reply:
The else is called if it's not found, not valid.  This PR does add logger so that we can pass it down, but it doesn't update all the ```std::cout <<``` log messages. All of these should be cleaned up.